lib/src/internal/http_endpoints.dart

```diff
@@ -1125,6 +1125,10 @@ class HttpEndpoints implements IHttpEndpoints {
 
   @override
   Future<IMessage> editMessage(Snowflake channelId, Snowflake messageId, MessageBuilder builder) async {
+    if (!builder.canBeUsedAsNewMessage()) {
+      return Future.error(ArgumentError("Cannot edit a message to have neither content nor embeds"));
+    }
```

Comment:
is that intended here? Checks says `canBeUsedAsNewMessage` and we are editing here

Reply:
Yes. The resulting message from the edit has to be a valid message, so `canBeUsedAsNewMessage` is checked.

It isn't perfect as some edge cases are not caught but it is the cleanest non-breaking solution, as other solutions would need the builder to keep some reference to the original message.